src/context/WalletProvider/NewWalletViews/sections/HardwareWalletsSection.tsx

```diff
@@ -0,0 +1,78 @@
+import { Box, Button, Flex, Stack, Text as CText, useColorModeValue } from '@chakra-ui/react'
+import { useCallback } from 'react'
+import { Text } from 'components/Text'
+import { KeyManager } from 'context/WalletProvider/KeyManager'
+import { LedgerConfig } from 'context/WalletProvider/Ledger/config'
+import { useWallet } from 'hooks/useWallet/useWallet'
+
+const Icon = LedgerConfig.icon
+
+const LedgerOption = ({
+  connect,
+  isSelected,
+  isDisabled,
+}: {
+  connect: () => void
+  isSelected: boolean
+  isDisabled: boolean
+}) => {
+  const backgroundColor = useColorModeValue('blackAlpha.100', 'whiteAlpha.100')
+
+  return (
+    <Box
+      as={Button}
+      key='ledger'
+      variant='ghost'
+      px={4}
+      ml={-4}
+      py={6}
+      borderRadius='md'
+      width='full'
+      onClick={connect}
+      bg={isSelected ? backgroundColor : undefined}
+      isDisabled={isDisabled}
+    >
+      <Flex alignItems='center' width='full'>
+        <Box boxSize='24px' mr={3}>
+          <Icon />
+        </Box>
+        <CText fontSize='md' fontWeight='medium'>
+          {LedgerConfig.name}
+        </CText>
+      </Flex>
+    </Box>
+  )
+}
+
+export const HardwareWalletsSection = ({
+  isLoading,
+  selectedWalletId,
+  onWalletSelect,
+}: {
+  isLoading: boolean
+  selectedWalletId: string | null
+  onWalletSelect: (id: string, initialRoute: string) => void
+}) => {
+  const { connect } = useWallet()
+
+  const handleConnectLedger = useCallback(() => {
+    onWalletSelect('ledger', '/ledger/connect')
+    connect(KeyManager.Ledger, false)
+  }, [connect, onWalletSelect])
+
+  return (
+    <Stack spacing={2} my={6}>
+      <Text
+        fontSize='sm'
+        fontWeight='medium'
+        color='gray.500'
+        translation='common.hardwareWallets'
+      />
+      <LedgerOption
+        connect={handleConnectLedger}
+        isSelected={selectedWalletId === 'ledger'}
```

Comment:
Is it feasible to have an enum of wallet id strings? Seems a little messy to have these string literals everywhere.

Reply:
We could, but not sure we should, these are *normally* programmatic (e.g rdns/native wallet ID).

Or well, maybe another way to see this is rdns/native wallet ID is the exception, and everything else is known of, in which case, think a better option would be to use `KeyManager` (the base) *or* rdns/native wallet ID.

Will take a note of that one and tackle in a follow-up as well since that relates to all and will be mergefix hell too on top of already existing mergefix hell 🙏🏽